pkg/volume/host_path/host_path.go

```diff
@@ -98,13 +98,47 @@ func (plugin *hostPathPlugin) GetAccessModes() []v1.PersistentVolumeAccessMode {
 	}
 }
 
-func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volume.VolumeOptions) (volume.Mounter, error) {
+func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, opts volume.VolumeOptions) (volume.Mounter, error) {
 	hostPathVolumeSource, readOnly, err := getVolumeSource(spec)
 	if err != nil {
 		return nil, err
 	}
+
+	path := hostPathVolumeSource.Path
+	// A containerized kubelet would have difficulty creating directories.
+	// The implementation will likely respect the containerized flag, allowing it to be "/rootfs/" aware and
+	// thus operate as desired.
+	if opts.Containerized {
+		path += "/rootfs" + path
+	}
+
+	_, err = os.Stat(path)
+	notExist := os.IsNotExist(err)
```

Comment:
This whole loop could be extracted to a function and unit tested with a type argument, and an interface to make dirs and files.  Does that make sense?  

```
type fileTypeChecker interface {
  Exists() bool
  IsFile() bool
  MakeFile() error
  IsDir() bool
  MakeDir() error
  IsBlock() bool
  IsChar() bool
  IsSocket() bool
}

type osFileTypeChecker struct {
  path string
  exists bool
  info os.FileInfo
}

func newOSFileTypeChecker(path string) (osFileTypeChecker, err) {
  ftc := osFileTypeChecker { path: path }
  if ftc.info, err := os.Stat(path); err != nil {
    if !os.IsNotExist(err) {
      return nil, err
    }
    ftc.exists = false
  }
  return ftc, nil
}

func (ftc osFileTypeChecker) Exists() bool {
  return ftc.exists
}

// and so on...

func checkType(path string, pathType HostPathType) error {
    ftc, err := newOSFileTypeChecker(path)
    if err != nil {
      return err
    }
    switch pathType {
    case Exists:
        if !ftc.Exists() {
          return fmt.Errorf("path %q does not exist")
        }
    case File:
        if !ftc.Exists() {
          return fmt.Errorf("path %q does not exist")
        }
        if !ftc.IsFile() {
          return fmt.Errorf("path %q is not a file")
        }
    /// and so on...
  }
  return nil
```